clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -915,10 +1163,12 @@ class AnnotatingParser {
             Previous->setType(TT_SelectorName);
           }
         }
-        if (CurrentToken->is(tok::colon) && OpeningBrace.is(TT_Unknown))
+        if (CurrentToken->is(tok::colon) && OpeningBrace.is(TT_Unknown) &&
+            !Style.isTableGen()) {
           OpeningBrace.setType(TT_DictLiteral);
-        else if (Style.isJavaScript())
+        } else if (Style.isJavaScript()) {
           OpeningBrace.overwriteFixedType(TT_DictLiteral);
+        }
```

Comment:
Line 1126 to here is inside parseBrace(), where is determining whether the l_brace is opening of dictionary literal.
TableGen does not have DictLitereals.